src/elements/types/JSXElement.js

```diff
@@ -0,0 +1,41 @@
+import Expression from '../Expression';
+
+export default class JSXElement extends Expression {
+    constructor(childNodes) {
+        super('JSXElement', childNodes);
+    }
+
+    _acceptChildren(children) {
+        let closingElement = null;
+        let childrenProperty = [];
+
+        let openingElement = children.passNode('JSXOpeningElement');
+        if (!openingElement.selfClosing) {
+            children.skipNonCode();
+            while (!children.isNode('JSXClosingElement')) {
+                childrenProperty.push(children.passOneOfNode(['Literal', 'JSXExpressionContainer', 'JSXElement']));
+            }
+            children.skipNonCode();
+            if (children.isNode('JSXClosingElement')) {
+                closingElement = children.passNode('JSXClosingElement');
```

Comment:
I think we can also check for the name of closing element (to be equal to opening element).